src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp

```diff
@@ -27,7 +28,14 @@ void CloudCollectorStartStopCaptureRequestWaiter::StartCapture(CaptureOptions ca
 CaptureServiceBase::StopCaptureReason
 CloudCollectorStartStopCaptureRequestWaiter::WaitForStopCaptureRequest() {
   absl::MutexLock lock(&stop_mutex_);
-  stop_mutex_.Await(absl::Condition(&stop_requested_));
+
+  if (max_capture_duration_.has_value()) {
```

Comment:
Isn't that the wrong way around? "If there is **NO** max_capture_duration set, then we wait without a deadline", right? What am I missing? :sweat_smile:

Reply:
If no max_capture_duration set, we will wait for the external stop request, i.e., someone call `CloudCollectorStartStopCaptureRequestWaiter::StopCapture`